hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapLoaderTest.java

```diff
@@ -170,61 +177,65 @@ public void testNullChecks_withMapStore_nullInKeys() {
+                        Map<String, String> val = new HashMap<String, String>();
+                        if (keys.contains("1")) {
+                            val.put("1", "1");
+                        }
+                        if (keys.contains("2")) {
+                            val.put(null, "2");
+                        }
+                        if (keys.contains("3")) {
+                            val.put("3", "3");
+                        }
+                        return val;
+                    }
+
+                    @Override
+                    public Iterable<String> loadAllKeys() {
+                        List<String> keys = new ArrayList<String>();
+                        keys.add("1");
+                        keys.add("2");
+                        keys.add("3");
+                        return keys;
+                    }
+                });
 
         Config config = new Config();
+        config.getMapConfig(name)
+                .setMapStoreConfig(mapStoreConfig);
 
-        MapConfig mapConfig = config.getMapConfig(name);
-        MapStoreConfig mapStoreConfig = new MapStoreConfig();
-        mapStoreConfig.setEnabled(true);
-        mapStoreConfig.setImplementation(new MapLoader() {
-            @Override
-            public Object load(Object key) {
-                if (key.equals("1")) {
-                    return "1";
-                }
-                if (key.equals("2")) {
-                    return "2";
-                }
-                if (key.equals("3")) {
-                    return "3";
-                }
-                return null;
-            }
-
-            @Override
-            public Map loadAll(Collection keys) {
-                Map val = new HashMap();
-                if (keys.contains("1")) {
-                    val.put("1", "1");
-                }
-                if (keys.contains("2")) {
-                    val.put(null, "2");
-                }
-                if (keys.contains("3")) {
-                    val.put("3", "3");
-                }
-                return val;
-            }
-
-            @Override
-            public Iterable loadAllKeys() {
-                List keys = new ArrayList();
-                keys.add("1");
-                keys.add("2");
-                keys.add("3");
-                return keys;
-            }
-        });
-        mapStoreConfig.setInitialLoadMode(MapStoreConfig.InitialLoadMode.LAZY);
-        mapConfig.setMapStoreConfig(mapStoreConfig);
         HazelcastInstance instance = createHazelcastInstance(config);
         IMap<String, String> map = instance.getMap(name);
 
-        expectedException.expect(NullPointerException.class);
-        expectedException.expectMessage(CoreMatchers.is("Key loaded by a MapLoader cannot be null."));
-        map.size();
+        try {
+            map.size();
+            fail("Expected a NPE due to a null key in a MapLoader");
```

Comment:
This test still fails spuriously! At least we see now, that no exception is thrown at all, not just a wrong message is being sent. I can see the exception being thrown and logged, so I guess sometimes it's just not thrown to the user correctly. Either there is a retry or it's swallowed somewhere.

@tombujok Can you please have a look?

Reply:
Actually this is https://github.com/hazelcast/hazelcast/issues/11244, which will not be fixed by this PR.

Reply:
I've added an `@Ignore` for this test, since the issue is put on the 3.10 milestone.